# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I chose **Lane 2: Refresh / Content Opportunity Scoring**.

The main goal of this lane is to identify which webpages should be reviewed first for actions such as refreshing, expanding, protecting, pruning, or monitoring content.

I chose this lane because content teams have limited time and cannot manually review every page with the same priority. A useful system should help rank pages based on observable search, traffic, engagement, and content signals so that reviewers can focus on the most promising or concerning pages first.

The starter dataset is also well suited to this problem because it contains information about impressions, clicks, sessions, content age, engagement, CTR, position, and performance trends.

This is not simply a "train a model" problem. The final goal is to improve the decision of **which pages should receive human attention first** and provide understandable reasons for that recommendation.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

I chose **Lane 2: Refresh / Content Opportunity Scoring**.

I chose this lane because content teams have limited time and cannot review every page equally. The goal is to rank webpages so reviewers can focus first on pages where the available search, traffic, engagement, and content signals suggest that review may be worthwhile.

This is not just a "train a model" problem. The real goal is to improve the decision of **which pages should receive human attention first** and eventually provide understandable reasons for that recommendation.

### Research question

Which webpages should be reviewed first for content refresh or other content actions based on their historical search, traffic, engagement, and content signals?

### Unit of analysis

The unit of analysis is **one webpage/content item**.

### Decision

The decision is **which pages a content or SEO team should review first when review capacity is limited**.

### Output

The intended output is a **ranked review queue**, where pages with stronger evidence of risk or opportunity appear higher in the list.

### Action

A reviewer can inspect a highly ranked page and decide whether to refresh outdated information, expand the content, improve metadata or intent match, protect the page, monitor it, or take no action.

The system recommends **where to look first**, not automatically what must be changed.

### Cost of a wrong call

A **false positive** means ranking a page highly when reviewing it provides little value. This wastes limited reviewer time and could lead to unnecessary changes.

A **false negative** means giving low priority to a page that actually deserved attention. This could cause the team to miss an important content or search opportunity.

### Why can data or ML help?

There are many interacting signals such as impressions, clicks, CTR, position, sessions, content age, freshness, and engagement. Data analysis and ML may help combine these signals to prioritize pages more effectively than looking at individual metrics manually.

The purpose of ML is therefore **decision support and ranking**, not simply training a model.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

I used the starter dataset to check whether the **Refresh / Content Opportunity Scoring** lane has enough evidence to justify further investigation.

I will look at the size of the content inventory, how many pages currently show a downward trend, and the amount of search visibility those pages receive.

In [7]:
import pandas as pd

# Load starter dataset directly from the FlyRank GitHub repository
url = "https://raw.githubusercontent.com/flyrank-bih/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

# Calculate supporting numbers
total_pages = len(df)

declining_pages = (df["trend_direction"] == "down").sum()
declining_pct = declining_pages / total_pages * 100

avg_impressions = df["impressions_90d"].mean()

print(f"Total pages: {total_pages:,}")
print(f"Pages with downward trend: {declining_pages:,}")
print(f"Percentage with downward trend: {declining_pct:.2f}%")
print(f"Average 90-day impressions: {avg_impressions:,.2f}")

Total pages: 30,000
Pages with downward trend: 16,262
Percentage with downward trend: 54.21%
Average 90-day impressions: 5,200.37


### What these numbers tell me

The starter dataset contains **30,000 pages**, which gives enough examples to investigate different patterns in content performance.

**16,262 pages (54.21%)** currently have a downward trend. This makes prioritization meaningful because manually reviewing every declining page would not be practical.

Pages also receive an average of approximately **5,200 impressions over 90 days**, showing that there is meaningful search visibility in the dataset.

Together, these numbers make this lane worth exploring over the next seven weeks. The useful problem is not simply identifying which pages are declining, but **ranking which pages deserve human review first based on the available evidence**.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify **observed patterns and directional signals** in historical search, traffic, engagement, and content data. It may help rank pages that deserve human review.

I can say that certain observable signals are associated with declining performance or appear useful for prioritization.

However, I **cannot claim that refreshing a page will cause its performance to improve**. The available data is observational, so a high-priority recommendation is not proof that a specific content change will produce a recovery.

I also cannot claim that the model discovers Google's ranking algorithm or that the observed features are confirmed Google ranking factors.

The current `trend_direction` field is useful as a starter proxy for decline, but it is not the same as predicting a future outcome. A stronger later version of the project could use historical features from one time window to predict decline or recovery in a separate future window.

Therefore, the final output should be treated as **decision support for human reviewers**, not an automatic content decision system.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.